CMakeLists.txt

```diff
@@ -16,18 +16,19 @@ set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 if (CMAKE_BUILD_TYPE STREQUAL "Debug")
-  set(ENABLE_ACC_CHECK ON)
+  set(IS_DEBUG_BUILD ON)
```

Comment:
Even though this variable was named differently before, we use different code to enable / disable diagnostic features (normal CMake `if()`) and for setting `CELERITY_DETAIL_ENABLE_DEBUG` (generator expressions, below). Why can't we set both with the simple `if()` that's being used here? We persist its result to Celerity installation anyway, so it does not matter whether the user applications build type actually matches the runtime's build type for this to be consistent.

Reply:
I *think* the reason why the `CELERITY_DETAIL_ENABLE_DEBUG` mechanism is so convoluted is due to multi-configuration builds (at the time I was still developing using Visual Studio). I guess we have to decide whether we want to support those or not.

Reply:
I like support for such things, but it would mean that we can't both have them as a standalone option and as something dependent on build type, right?